theopensystemslab/planx-new
# fix: don't sync teams.settings data from production, use default column value

`teams.settings.sendToEmail` is now populated with council emails on production. We want to be able to freely test send to email locally and on pizzas without fear of accidently emailing someone, therefore we do want to include this value in our data sync. 

Since `teams.settings` column has a default value for `externalPlanningSite`, I think we can just fully omit syncing this column for right now. I don't know any clever GraphQL query syntax for omitting a jsonb key, only for selecting a specific key. By omitting settings altogether and only using the default column value, we'll lose `sendToEmail` (good!) and occassionally `homepage` properties (which means logos won't link to council websites on pizzas), but I think that's an okay trade-off right now!

Ideas/next steps for more flexibility are:
- add a new text column to `teams` (or flows even, but I don't think there's actually a use-case for this yet) to store `sendToEmail` rather than as a jsonb property on `settings`, so it's easier to omit from the data sync independently
- move `sendToEmail` into team or flow level settings in the Editor

Also worth noting: 
- It'd be most ideal to be able to sync `sendToEmail` from production to staging (more & more POs are testing on .dev), but _not_ to docker environments (local and pizza), but this data sync script is shared between all envs right now

To test:
- Open Hasura `teams` table and confirm settings column is only populated with default `externalPlanningSite` key
